fl4health/utils/dataset_converter.py

```diff
@@ -0,0 +1,159 @@
+from abc import ABC
+from typing import Callable, Optional, Tuple, Union
+
+import torch
+
+from fl4health.utils.dataset import BaseDataset
+
+
+class DatasetConverter(ABC, BaseDataset):
+    """Dataset converter classes are designed to re-format any dataset for a given training task,
+    and to fit it into the unified training scheme of supervised learning in clients.
+    Converters can be used in the data loading step. They can also apply a light
+    pre-processing step on datasets before the training process.
+    """
+
+    def __init__(self, converter_function: Callable, dataset: Union[None, BaseDataset]) -> None:
```

Comment:
I think we can be more descriptive with the `converter_function` typing here. That is 
```
converter_function: Callable[[torch.Tensor, torch.Tensor], Tuple[torch.Tensor, torch.Tensor]]
```
It's a bit verbose, so you could also define a type alias, but it allows for mypy to be more precise in its type enforcement